src/core/border_router/routing_manager.hpp

```diff
@@ -243,18 +289,74 @@ class RoutingManager : public InstanceLocator
     static void HandleRouterSolicitTimer(Timer &aTimer);
     void        HandleRouterSolicitTimer(void);
 
-    static void HandleDiscoveredOnLinkPrefixInvalidTimer(Timer &aTimer);
-    void        HandleDiscoveredOnLinkPrefixInvalidTimer(void);
+    static void HandleDiscoveredPrefixInvalidTimer(Timer &aTimer);
+    void        HandleDiscoveredPrefixInvalidTimer(void);
 
     void HandleRouterSolicit(const Ip6::Address &aSrcAddress, const uint8_t *aBuffer, uint16_t aBufferLength);
     void HandleRouterAdvertisement(const Ip6::Address &aSrcAddress, const uint8_t *aBuffer, uint16_t aBufferLength);
 
+    /**
+     * This method updates the discovered prefix list with given Prefix Information Option.
+     *
+     * @param[in]  aPio  A reference to the Prefix Information Option.
+     *
+     * @returns  TRUE if the prefix list was updated, FALSE otherwise.
+     *
+     */
+    bool UpdateDiscoveredPrefixes(const RouterAdv::PrefixInfoOption &aPio);
```

Comment:
Just a suggestion (please feel free to leave for maybe later PRs): 
I noticed that you have nice documentation for all these methods (and many more from earlier PR), but these methods are all `private`. I think it may be better not to use the doxygen format for the private method so that they are not included in the doxygen API documentations). My suggestion like before is to move them in `cpp` file. Again, it's fine for me as is (just a suggestion for later maybe).

Reply:
Thanks, updated to the suggested style in new push, please review.